store/sqlstore/user_store.go

```diff
@@ -1255,7 +1255,31 @@ func (us SqlUserStore) AnalyticsActiveCount(timePeriod int64, options model.User
 
 	v, err := us.GetReplica().SelectInt(queryStr, args...)
 	if err != nil {
-		return 0, model.NewAppError("SqlUserStore.AnalyticsDailyActiveUsers", "store.sql_user.analytics_daily_active_users.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return 0, model.NewAppError("SqlUserStore.AnalyticsActiveCount", "store.sql_user.analytics_daily_active_users.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return v, nil
+}
+
+func (us SqlUserStore) AnalyticsActiveCountForPeriod(startTime int64, endTime int64, options model.UserCountOptions) (int64, *model.AppError) {
```

Comment:
I miss some tests around the new method.

Reply:
We are trying to return plain errors from the store now, do you mind to return `error` instead of `*model.AppError`, and replace the `model.NewAppError` to regular errors? Because all of them are considered `InternalServerErrors` you only have to return plain standar errors, can be just `errors.Wrap(err, "error description")` (probably you can see examples of this in other stores.

Reply:
Thanks